editor/src/clj/editor/game_object.clj

```diff
@@ -254,18 +260,29 @@
 
 ;; -----------------------------------------------------------------------------
 
+(defn- get-all-comp-exts [workspace]
+  (keep (fn [[ext {:keys [tags :as _resource-type]}]]
+          (when (or (contains? tags :component)
+                    (contains? tags :embeddable))
+            ext))
+        (workspace/get-resource-type-map workspace)))
+
 (g/defnode ReferencedComponent
   (inherits ComponentNode)
 
   (property path g/Any
-            (dynamic edit-type (g/fnk [source-resource]
-                                      {:type resource/Resource
-                                       :ext (some-> source-resource resource/resource-type :ext)
-                                       :to-type (fn [v] (:resource v))
-                                       :from-type (fn [r] {:resource r :overrides []})}))
+            (dynamic edit-type (g/fnk [source-resource _node-id]
+                                 (let [resource-type (some-> source-resource resource/resource-type)
+                                       tags (:tags resource-type)]
+                                   {:type resource/Resource
+                                    :ext (or (and (or (contains? tags :component) (contains? tags :embeddable))
+                                                  (some-> resource-type :ext))
+                                             (get-all-comp-exts (project/workspace (project/get-project _node-id))))
```

Comment:
This is strange if we show an error "this is not a component" but right after that suggest picking another one "non-component".  This fix adds a filter to "only components and embeddables" in cases when non-component was picked.